WebGoatCore/Data/ProductRepository.cs

```diff
@@ -33,12 +33,12 @@ public List<Product> GetTopProducts(int numberOfProductsToReturn)
                 .Take(numberOfProductsToReturn)
                 .ToList();
 
-            if (topProducts.Count == 0)
-            {
-                topProducts = _context.Products
+            if(topProducts.Count < 4)
+            {
+                topProducts.AddRange(_context.Products
                     .OrderByDescending(p => p.UnitPrice)
```

Comment:
It specifies different criteria than topProducts taken above. It is not consistent.